RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -56,4 +76,29 @@ private String convertToBulkRequestBody(List<String> bulkSections) {
         }
         return builder.toString();
     }
+
+    private static java.util.function.Predicate<String> statefulSizeSegmentingPredicate(int maxItems, long maxSizeInBytes) {
+        return new java.util.function.Predicate<>() {
+            private int currentItemCount = 0;
+            private long currentSize = 0;
+
+            @Override
+            public boolean test(String next) {
```

Comment:
nit: You could do this w/ just a lambda if you pulled lines 82 & 83 out into the local statefulSizeSegmentingPredicate scope, right?

Reply:
Moved this, not sure if it looks cleaner